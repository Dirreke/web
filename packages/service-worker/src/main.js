import { WebDav } from './webDav.js'
importScripts("https://storage.googleapis.com/workbox-cdn/releases/4.3.1/workbox-sw.js");

console.log('%cðŸ”¨ ServiceWorker initialized ', 'background: #273d3d; color: white');

// auto generate from webpack manifest
workbox.precaching.precacheAndRoute(self.__WB_MANIFEST, {
  // Ignore all URL parameters.
  ignoreURLParametersMatching: [/.*/] // main.js is loaded with a version hash
});

self.addEventListener('install', e => {
  self.skipWaiting();
});

const client = new WebDav()
addEventListener('message', async (event) => {
  if (event.data.type === 'health') {
    event.ports[0].postMessage(true);
    console.log('%cðŸ”¨ ServiceWorker up and running ', 'background: green; color: white');
  }
  if (event.data.type === 'copy') {
    const data = event.data
    const sourceSpaceId = data.sourceSpaceId
    const sourcePath = data.sourcePath
    const targetSpaceId = data.targetSpaceId
    const targetPath = data.targetPath
    const token = data.token

    console.log(`%cðŸ”¨ ServiceWorker copy file from ${sourcePath} to ${targetPath}`, 'background: blue; color: white');
    await client.moveFile(sourceSpaceId, sourcePath, targetSpaceId, targetPath, token)
  }
});

