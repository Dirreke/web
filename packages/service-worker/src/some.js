export class Client {
    test() {
        console.log("LOL IM A MODULE")
    }
}