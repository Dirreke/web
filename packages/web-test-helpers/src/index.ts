export * from './mocks/defaultComponentMocks'
export * from './mocks/defaultStubs'
export * from './mocks/axios'

export * from './defaultPlugins'
export * from './helpers'
