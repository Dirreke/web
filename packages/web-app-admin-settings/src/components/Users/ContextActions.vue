<template>
  <div>
    <context-action-menu :menu-sections="menuSections" :action-options="{ resources: items }" />
    <quota-modal
      v-if="quotaModalIsOpen"
      :cancel="closeQuotaModal"
      :spaces="selectedPersonalDrives"
      :max-quota="maxQuota"
      resource-type="user"
      @space-quota-updated="spaceQuotaUpdated"
    />
  </div>
</template>

<script lang="ts">
import ShowDetails from '../../mixins/showDetails'
import Delete from '../../mixins/users/delete'
import Edit from '../../mixins/users/edit'
import {
  computed,
  defineComponent,
  getCurrentInstance,
  PropType,
  unref,
  watch,
  toRaw,
  ref
} from 'vue'
import ContextActionMenu from 'web-pkg/src/components/ContextActions/ContextActionMenu.vue'
import { User } from 'web-client/src/generated'
import QuotaModal from 'web-pkg/src/components/Spaces/QuotaModal.vue'
import { useSpaceActionsEditQuota } from 'web-pkg/src/composables/actions/spaces'
import { SpaceResource } from 'web-client/src'
import { useCapabilitySpacesMaxQuota, useStore } from 'web-pkg/src/composables'
import { SpaceActionOptions } from 'web-pkg/src/composables/actions'

export default defineComponent({
  name: 'ContextActions',
  components: { ContextActionMenu, QuotaModal },
  mixins: [Delete, Edit, ShowDetails],
  props: {
    items: {
      type: Array as PropType<User[]>,
      required: true
    }
  },
  setup(props) {
    const instance = getCurrentInstance().proxy as any
    const store = useStore()
    const filterParams = computed(() => ({ resources: props.items }))
    const selectedPersonalDrives = ref([])
    watch(
      () => props.items,
      async () => {
        selectedPersonalDrives.value.splice(0, unref(selectedPersonalDrives).length)
        props.items.forEach((user) => {
          const drive = toRaw(user.drive)
          if (drive === undefined || drive.id === undefined) {
            return
          }
          const spaceResource = {
            id: drive.id,
            name: user.displayName,
            spaceQuota: drive.quota
          } as SpaceResource
          selectedPersonalDrives.value.push(spaceResource)
        })
      },
      { deep: true, immediate: true }
    )

    const {
      actions: editQuotaActions,
      modalOpen: quotaModalIsOpen,
      closeModal: closeQuotaModal,
      spaceQuotaUpdated
    } = useSpaceActionsEditQuota({ store })

    const menuItemsPrimaryActions = computed(() =>
      [...instance.$_edit_items, ...instance.$_delete_items].filter((item) =>
        item.isEnabled(unref(filterParams))
      )
    )
    const menuItemsSecondaryActions = computed(() =>
      [...unref(editQuotaActions)].filter((item) =>
        // Why does this work at all? Edit Quota is a space action and props.items is of type User[]
        item.isEnabled(unref(filterParams) as SpaceActionOptions)
      )
    )

    const menuItemsSidebar = computed(() =>
      [...instance.$_showDetails_items].filter((item) => item.isEnabled(unref(filterParams)))
    )

    const menuSections = computed(() => {
      const sections = []

      if (unref(menuItemsPrimaryActions).length) {
        sections.push({
          name: 'primaryActions',
          items: unref(menuItemsPrimaryActions)
        })
      }
      if (unref(menuItemsSecondaryActions).length) {
        sections.push({
          name: 'secondaryActions',
          items: unref(menuItemsSecondaryActions)
        })
      }
      if (unref(menuItemsSidebar).length) {
        sections.push({
          name: 'sidebar',
          items: unref(menuItemsSidebar)
        })
      }
      return sections
    })

    return {
      maxQuota: useCapabilitySpacesMaxQuota(),
      menuSections,
      quotaModalIsOpen,
      closeQuotaModal,
      spaceQuotaUpdated,
      selectedPersonalDrives
    }
  }
})
</script>
