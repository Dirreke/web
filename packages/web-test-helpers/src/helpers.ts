import { defaultLocalVue } from 'web-test-helpers/src/localVue/defaultLocalVue'
import Vuex, { StoreOptions } from 'vuex'
import { Component } from 'vue'
import { mount as _mount } from '@vue/test-utils'
import { Data, defineComponent, SetupFunction } from '@vue/composition-api'
import { defaultPlugins } from 'web-test-helpers'
export { RouterLinkStub } from '@vue/test-utils'

export const createStore = <T>(storeOptions: StoreOptions<T>) => {
  return {
    install(app) {
      app.prototype.$store = new Vuex.Store(storeOptions)
    }
  }
}

type ComponentType = any

type CompatPlugin = any

type CompatGlobalMountOptions = {
  mocks?: object | false
  stubs?:
    | {
        [key: string]: Component | string | boolean
      }
    | string[]
  attrs?: Record<string, unknown>
  plugins?: (CompatPlugin | [CompatPlugin, ...any[]])[]
  directives?: { [key: string]: any }
  provide?: Record<any, any>
  components?: Record<string, ComponentType>
}

type CompatMountOptions = {
  props?: any
  data?: any
  attachTo?: HTMLElement | string
  global?: CompatGlobalMountOptions
  shallow?: boolean
  slots?: any
}

export const mount = (component: ComponentType, options: CompatMountOptions) => {
  const localVue = defaultLocalVue()

  options?.global?.plugins?.filter((p) => p).forEach(localVue.use)

  for (const [name, component] of Object.entries(options?.global?.components || {})) {
    localVue.component(name, component)
  }

  return _mount(component, {
    localVue,
    ...(options?.shallow && { shouldProxy: options.shallow }),
    ...(options?.props && { propsData: options.props }),
    ...(options?.data && { data: options.data }),
    ...(options?.attachTo && { attachTo: options.attachTo }),
    ...(options?.slots && { slots: options.slots }),
    ...(options?.global?.provide && { provide: options.global.provide }),
    ...(options?.global?.mocks && { mocks: options.global.mocks }),
    ...(options?.global?.stubs && { stubs: options.global.stubs })
  })
}

export const shallowMount = (component: ComponentType, options: CompatMountOptions) => {
  options = options || {}
  options.shallow = true

  return mount(component, options)
}

export const getComposableWrapper = (
  setup: SetupFunction<Data, Data>,
  { mocks = undefined, store = undefined } = {}
) => {
  return mount(
    defineComponent({
      setup,
      template: '<div></div>'
    }),
    {
      global: {
        plugins: [...defaultPlugins(), store],
        ...(mocks && { mocks })
      }
    }
  )
}
