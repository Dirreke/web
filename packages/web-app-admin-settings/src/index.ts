import translations from '../l10n/translations.json'
import General from './views/General.vue'
import Users from './views/Users.vue'
import Groups from './views/Groups.vue'
import Spaces from './views/Spaces.vue'
import { Ability } from 'web-pkg'

// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

const appInfo = {
  name: $gettext('Administration Settings'),
  id: 'admin-settings',
  icon: 'settings-4',
  isFileEditor: false
}

const routes = ({ $ability }: { $ability: Ability }) => [
  {
    path: '/',
    redirect: () => {
      if ($ability.can('read-all', 'Setting')) {
        return { name: 'admin-settings-general' }
      }
      if ($ability.can('read-all', 'Account')) {
        return { name: 'admin-settings-users' }
      }
      if ($ability.can('read-all', 'Space')) {
        return { name: 'admin-settings-spaces' }
      }
      throw Error('Insufficient permissions')
    }
  },
  {
    path: '/general',
    name: 'admin-settings-general',
    component: General,
    meta: {
      authContext: 'user',
      title: $gettext('General')
    }
  },
  {
    path: '/users',
    name: 'admin-settings-users',
    component: Users,
    meta: {
      authContext: 'user',
      title: $gettext('Users')
    }
  },
  {
    path: '/groups',
    name: 'admin-settings-groups',
    component: Groups,
    meta: {
      authContext: 'user',
      title: $gettext('Groups')
    }
  },
  {
    path: '/spaces',
    name: 'admin-settings-spaces',
    component: Spaces,
    meta: {
      authContext: 'user',
      title: $gettext('Spaces')
    }
  }
]

const navItems = ({ $ability }: { $ability: Ability }) => [
  {
    name: $gettext('General'),
    icon: 'settings-4',
    route: {
      path: `/${appInfo.id}/general?`
    },
    enabled: () => {
      return $ability.can('read-all', 'Setting')
    }
  },
  {
    name: $gettext('Users'),
    icon: 'user',
    route: {
      path: `/${appInfo.id}/users?`
    },
    enabled: () => {
      return $ability.can('read-all', 'Account')
    }
  },
  {
    name: $gettext('Groups'),
    icon: 'group-2',
    route: {
      path: `/${appInfo.id}/groups?`
    },
    enabled: () => {
      return $ability.can('read-all', 'Group')
    }
  },
  {
    name: $gettext('Spaces'),
    icon: 'layout-grid',
    route: {
      path: `/${appInfo.id}/spaces?`
    },
    enabled: () => {
      return $ability.can('read-all', 'Space')
    }
  }
]

export default {
  appInfo,
  routes,
  translations,
  navItems
}
