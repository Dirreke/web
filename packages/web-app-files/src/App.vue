<template>
  <main id="files" class="oc-flex oc-height-1-1">
    <div v-if="dragareaEnabled" class="dragarea" />
    <router-view tabindex="0" class="oc-width-expand" />
  </main>
</template>
<script lang="ts">
import { defineComponent, watch } from 'vue'
import { useRoute, useStore } from 'web-pkg/src/composables'
import { eventBus } from 'web-pkg/src/services/eventBus'
import { mapMutations } from 'vuex'
import { buildResource } from 'web-client/src/helpers'

export default defineComponent({
  setup() {
    const store = useStore<any>()
    watch(useRoute(), () => {
      store.dispatch('Files/resetFileSelection')
    })
  },
  data: () => ({
    dragareaEnabled: false
  }),
  created() {
    const dragOver = eventBus.subscribe('drag-over', this.onDragOver)
    const dragOut = eventBus.subscribe('drag-out', this.hideDropzone)
    const drop = eventBus.subscribe('drop', this.hideDropzone)
    const serviceWorkerEdit = eventBus.subscribe('sw.copy', this.onServiceWorkerCopy)

    this.$on('beforeDestroy', () => {
      eventBus.unsubscribe('drag-over', dragOver)
      eventBus.unsubscribe('drag-out', dragOut)
      eventBus.unsubscribe('drop', drop)
    })
  },

  methods: {
    ...mapMutations('Files', ['UPSERT_RESOURCE']),

    hideDropzone() {
      this.dragareaEnabled = false
    },
    onDragOver(event) {
      this.dragareaEnabled = (event.dataTransfer.types || []).some((e) => e === 'Files')
    },
    onServiceWorkerCopy(event) {
      const response = event.response
      const resource = buildResource(response)
      console.log(resource)
      //context.commit('UPSERT_RESOURCE', movedResource)
      //this.UPSERT_RESOURCE(resource)
      console.log("SERVICE WORKER COPY EVENT", event)
    }
  }
})
</script>

<style lang="scss" scoped>
main {
  max-height: 100%;
}

.dragarea {
  background-color: rgba(60, 130, 225, 0.21);
  pointer-events: none;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  position: absolute;
  z-index: 9;
  border-radius: 14px;
  border: 2px dashed var(--oc-color-swatch-primary-muted);
}

#files {
  position: relative;
}
</style>
