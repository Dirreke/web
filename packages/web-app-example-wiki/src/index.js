import { bus } from 'web-pkg/src/instance'
import { WikiSearch } from './search'
import App from './App.vue'
import Page from './views/Page.vue'

const appInfo = {
  name: 'example-app-wiki',
  id: 'wiki',
  icon: 'folder',
  isFileEditor: true,
  extensions: []
}

export default {
  appInfo,
  routes: [
    {
      name: 'wiki',
      path: '/',
      components: {
        app: App
      },
      children: [
        {
          name: 'page',
          path: 'page/:id',
          component: Page
        }
      ]
    }
  ],
  mounted({ router }) {
    bus.emit('app.search.register.provider', new WikiSearch(router))
  }
}
