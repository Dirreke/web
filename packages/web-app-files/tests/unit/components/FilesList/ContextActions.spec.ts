import {
  createStore,
  defaultComponentMocks,
  defaultPlugins,
  defaultStubs,
  mount
} from 'web-test-helpers'
import { mockDeep } from 'jest-mock-extended'
import { SpaceResource } from 'web-client/src/helpers'
import { defaultStoreMockOptions } from 'web-test-helpers/src/mocks/store/defaultStoreMockOptions'
import ContextActions from 'web-app-files/src/components/FilesList/ContextActions.vue'
import fixtureMimeTypes from 'web-app-files/tests/__fixtures__/mimeTypes.js'

const exampleApps = [
  { name: 'exampleName1', icon: 'exampleIcon1' },
  { name: 'exampleName2', icon: 'exampleIcon2' },
  { name: 'exampleName3', icon: 'exampleIcon3' }
]

const mockMenuSections = [
  {
    name: 'context',
    items: [
      {
        icon: 'file-text',
        canBeDefault: true,
        handler: jest.fn(),
        label: () => 'Open in editor',
        componentType: 'button',
        selector: '.oc-files-actions-text-editor-trigger',
        class: 'oc-files-actions-text-editor-trigger'
      },
      ...exampleApps.map((app) => {
        return {
          img: app.icon,
          canBeDefault: true,
          handler: jest.fn(),
          label: () => 'Open in ' + app.name,
          componentType: 'button',
          selector: `.oc-files-actions-${app.name}-trigger`,
          class: `oc-files-actions-${app.name}-trigger`
        }
      }),
      {
        icon: 'download',
        canBeDefault: true,
        handler: jest.fn(),
        label: () => 'Download',
        componentType: 'button',
        selector: '.oc-files-actions-download-file-trigger',
        class: 'oc-files-actions-download-file-trigger'
      },
      {
        icon: 'links',
        handler: jest.fn(),
        label: () => 'Create link',
        componentType: 'button',
        selector: '.oc-files-actions-create-public-link-trigger',
        class: 'oc-files-actions-create-public-link-trigger'
      },
      {
        icon: 'group',
        handler: jest.fn(),
        label: () => 'Share',
        componentType: 'button',
        selector: '.oc-files-actions-show-shares-trigger',
        class: 'oc-files-actions-show-shares-trigger'
      }
    ]
  },
  {
    name: 'actions',
    items: [
      {
        icon: 'pencil',
        handler: jest.fn(),
        label: () => 'Edit',
        componentType: 'button',
        selector: '.oc-files-actions-rename-trigger',
        class: 'oc-files-actions-rename-trigger'
      },
      {
        icon: 'folder-shared',
        handler: jest.fn(),
        label: () => 'Move',
        componentType: 'button',
        selector: '.oc-files-actions-move-trigger',
        class: 'oc-files-actions-move-trigger'
      },
      {
        icon: 'file-copy',
        handler: jest.fn(),
        label: () => 'Copy',
        componentType: 'button',
        selector: '.oc-files-actions-copy-trigger',
        class: 'oc-files-actions-copy-trigger'
      },
      {
        icon: 'delete-bin-5',
        handler: jest.fn(),
        label: () => 'Delete',
        componentType: 'button',
        selector: '.oc-files-actions-delete-trigger',
        class: 'oc-files-actions-delete-trigger'
      },
      {
        icon: 'slideshow',
        handler: jest.fn(),
        label: () => 'All actions',
        componentType: 'button',
        selector: '.oc-files-actions-show-actions-trigger',
        class: 'oc-files-actions-show-actions-trigger'
      }
    ]
  },
  {
    name: 'sidebar',
    items: [
      {
        icon: 'information',
        handler: jest.fn(),
        label: () => 'Details',
        componentType: 'button',
        selector: '.oc-files-actions-show-details-trigger',
        class: 'oc-files-actions-show-details-trigger'
      }
    ]
  }
]

const filesPersonalRoute = { name: 'files-personal' }

describe('ContextActions', () => {
  // FIXME: Fix test
  // describe('action handlers', () => {
  //   afterEach(() => {
  //     jest.clearAllMocks()
  //   })
  //   it('renders action handlers as clickable elements', async () => {
  //     const { wrapper } = getWrapper(
  //       filesPersonalRoute,
  //       {
  //         name: 'exampleFile',
  //         extension: 'jpg',
  //         mimeType: 'application/fileFormat2',
  //         type: 'file'
  //       },
  //       exampleApps
  //     )
  //
  //     console.log(wrapper.html())
  //     for (const section of mockMenuSections) {
  //       for (const item of section.items) {
  //         const buttonElement = wrapper.find(item.selector)
  //         expect(buttonElement.exists()).toBeTruthy()
  //         await buttonElement.trigger('click')
  //         expect(item.handler).toHaveBeenCalledTimes(1)
  //       }
  //     }
  //   })
  // })

  describe('menu items', () => {
    it('renders a list of actions for a file', () => {
      const { wrapper } = getWrapper(filesPersonalRoute, {
        name: 'exampleFile',
        extension: 'jpg',
        mimeType: 'application/fileFormat2',
        type: 'file'
      })

      expect(wrapper).toMatchSnapshot()
    })

    it('renders a list of actions for a folder', () => {
      const { wrapper } = getWrapper(filesPersonalRoute, {
        name: 'exampleFolder',
        extension: '',
        type: 'folder'
      })

      expect(wrapper).toMatchSnapshot()
    })
  })
})

function getWrapper(route, { filename, extension, type = '', mimeType }: any, availableApps = []) {
  const storeOptions = {
    ...defaultStoreMockOptions,
    state: {
      apps: { fileEditors: [] },
      user: { capabilities: {} }
    },
    External: {
      namespaced: true,
      getters: {
        mimeTypes: () => {
          return fixtureMimeTypes
        }
      }
    },
    getters: {
      ...defaultStoreMockOptions.getters,
      capabilities: () => ({
        files: {
          app_providers: [
            {
              apps_url: '/app/list',
              enabled: true,
              open_url: '/app/open'
            }
          ]
        }
      })
    }
  }
  storeOptions.modules.Files.getters.currentFolder.mockImplementation(() => '/')
  const store = createStore(storeOptions)
  const mocks = defaultComponentMocks()

  return {
    storeOptions,
    mocks,
    wrapper: mount(ContextActions, {
      data: () => ({ appList: availableApps }),
      props: {
        space: mockDeep<SpaceResource>(),
        items: [
          {
            id: 'a93f8adf==',
            fileId: 'a93f8adf==',
            name: filename,
            path: type === 'file' ? `/${filename}.${extension}` : `/${filename}`,
            mimeType,
            extension,
            type,
            canDownload: () => true,
            isReceivedShare: () => true,
            canBeDeleted: () => true,
            canRename: () => true
          }
        ]
      },
      global: {
        mocks,
        provide: { currentSpace: mockDeep<SpaceResource>() },
        stubs: defaultStubs,
        plugins: [...defaultPlugins(), store]
      }
    })
  }
}
