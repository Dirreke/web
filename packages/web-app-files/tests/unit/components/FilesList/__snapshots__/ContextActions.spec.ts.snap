// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextActions menu items renders a list of actions for a file 1`] = `
<div id="oc-files-context-menu">
  <ul id="oc-files-context-actions-actions" class="oc-list oc-my-rm oc-mx-rm oc-files-context-actions">
    <li class="oc-files-context-action oc-px-s oc-rounded oc-menu-item-hover"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-download-file-trigger action-menu-item" data-testid="action-handler"><span class="oc-icon oc-icon-m oc-icon-passive" data-testid="action-icon"><!----></span> <span data-testid="action-label" class="oc-files-context-action-label">Download</span>
        <!---->
        <!---->
      </button></li>
    <li class="oc-files-context-action oc-px-s oc-rounded oc-menu-item-hover"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-rename-trigger action-menu-item" data-testid="action-handler"><span class="oc-icon oc-icon-m oc-icon-passive" data-testid="action-icon"><!----></span> <span data-testid="action-label" class="oc-files-context-action-label">Rename</span>
        <!---->
        <!---->
      </button></li>
  </ul>
</div>
`;

exports[`ContextActions menu items renders a list of actions for a folder 1`] = `
<div id="oc-files-context-menu">
  <ul id="oc-files-context-actions-actions" class="oc-list oc-my-rm oc-mx-rm oc-files-context-actions">
    <li class="oc-files-context-action oc-px-s oc-rounded oc-menu-item-hover"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-download-file-trigger action-menu-item" data-testid="action-handler"><span class="oc-icon oc-icon-m oc-icon-passive" data-testid="action-icon"><!----></span> <span data-testid="action-label" class="oc-files-context-action-label">Download</span>
        <!---->
        <!---->
      </button></li>
    <li class="oc-files-context-action oc-px-s oc-rounded oc-menu-item-hover"><button type="button" class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-rename-trigger action-menu-item" data-testid="action-handler"><span class="oc-icon oc-icon-m oc-icon-passive" data-testid="action-icon"><!----></span> <span data-testid="action-label" class="oc-files-context-action-label">Rename</span>
        <!---->
        <!---->
      </button></li>
  </ul>
</div>
`;
